kolibri/plugins/coach/assets/src/composables/useQuizCreation.js

```diff
@@ -414,15 +418,27 @@ export default function useQuizCreation(DEBUG = false) {
   );
   /** @type {ComputedRef<QuizQuestion[]>} All questions in the active section's `resource_pool`
    *                                      exercises */
-  const activeQuestionsPool = computed(() => []);
+  const activeQuestionsPool = computed(() => {
+    const pool = get(activeResourcePool);
+    return selectQuestions(
+      pool.reduce((count, r) => count + r.assessmentmetadata.assessment_item_ids.length, 0),
+      pool.map(r => r.content_id),
+      pool.map(r => r.title),
+      pool.map(r => r.assessmentmetadata.assessment_item_ids),
+      get(_quiz).seed
+    );
+  });
   /** @type {ComputedRef<QuizQuestion[]>} All questions in the active section's `questions` property
    *                                      those which are currently set to be used in the section */
   const activeQuestions = computed(() => get(activeSection).questions);
   /** @type {ComputedRef<String[]>} All question_ids the user has selected for the active section */
   const selectedActiveQuestions = computed(() => get(_selectedQuestionIds));
   /** @type {ComputedRef<QuizQuestion[]>} Questions in the active section's `resource_pool` that
    *                                         are not in `questions` */
-  const replacementQuestionPool = computed(() => {});
+  const replacementQuestionPool = computed(() => {
```

Comment:
Sorry, I didn't spot this before, but this is important for data validity and avoiding weird bugs.

Code here seems to be assuming that `question_id` is a globally unique property of a question, whereas it is only guaranteed to be unique _within_ an exercise.

To specifically select a question, we need to specify both the question_id and the exercise_id - this is why for coach quizzes, we create a synthetic item id out of exercise_id and question_id: (nodeId and itemId) https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/learn/assets/src/views/ExamPage/index.vue#L391

We should use a similar synthetic identifier for checking whether questions have been selected or not.

Noting that this is a fairly constrained issue in terms of impact, as we are recording both the nodeId and the itemId as exercise_id and question_id in the exam data, so it's only for doing these crosschecks we need to be careful.

Reply:
Thanks Richard - to be sure I'm thinking about this correctly:

`exercise_id` - (AKA `ContentNode.id`?) - refers to any possible version of an Exercise resource and is consistent for that resource (aka ContentNode?)

`question_id` - (AKA `assessment_item_id`?) - refers to a unique question within a _particular version_ of the exercise and may be (or always will be?) different from previous versions of the same question

I'm not seeing exactly where using the `question_id` in this case results in a potential issue though for finding potential replacements so maybe I'm misunderstanding something above.

Reply:
Are we only allowing replacements from within the same content node? My assumption was that we can replace any question with any other question from the selected pool of exercises.

If this is the case, we might have selected two different exercises that have different contentnode ids, but have one or more assessment_item_ids that are the same, so that is where we could get an issue if we do not uniquely identify them by the combination of exercise_id and question_id.

Reply:
This has now penetrated into my brain for sure -- I think that perhaps I will make that into a follow-up issue, perhaps it can be well informed by a data architecture review as well!